okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/UrlSelectorImpl.java

```diff
@@ -30,17 +31,19 @@
 import java.util.Optional;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Supplier;
 import okhttp3.HttpUrl;
 
 final class UrlSelectorImpl implements UrlSelector {
 
-    private final ImmutableList<HttpUrl> baseUrls;
+    private final Supplier<List<HttpUrl>> baseUrls;
     private final AtomicInteger currentUrl;
     private final Cache<HttpUrl, UrlAvailability> failedUrls;
     private final boolean useFailedUrlCache;
 
-    private UrlSelectorImpl(ImmutableList<HttpUrl> baseUrls, Duration failedUrlCooldown) {
-        this.baseUrls = baseUrls;
+    private UrlSelectorImpl(ImmutableList<HttpUrl> baseUrls, boolean randomizeOrder, Duration failedUrlCooldown) {
+        this.baseUrls = Suppliers.memoizeWithExpiration(
+                () -> randomize(baseUrls, randomizeOrder), 10, TimeUnit.MINUTES);
```

Comment:
We create most of our clients at the same time when a service is started. Assuming relatively heavy use, expiration will trigger almost exactly every ten minutes. This means every ten minutes we should expect degraded performance and higher error rates as all clients simultaneously rotate to new nodes.

Reply:
Updated to include some jitter in the period that we refresh at